src/space-picker/space-picker.js

```diff
@@ -0,0 +1,127 @@
+import ColorElement from "../common/color-element.js";
+
+const Self = class SpacePicker extends ColorElement {
+	static tagName = "space-picker";
+	static url = import.meta.url;
+	static shadowStyle = true;
+	static shadowTemplate = `<select id="picker" part="picker"></select>`;
+
+	constructor () {
+		super();
+
+		this._el = {};
+		this._el.picker = this.shadowRoot.querySelector("#picker");
+	}
+
+	connectedCallback () {
+		super.connectedCallback?.();
+		this._el.picker.addEventListener("change", this);
+	}
+
+	disconnectedCallback () {
+		super.disconnectedCallback?.();
+		this._el.picker.removeEventListener("change", this);
+	}
+
+	handleEvent (event) {
+		if (event.type === "change" && event.target === this._el.picker) {
+			this.value = this._el.picker.value;
+		}
+
+		this.dispatchEvent(new event.constructor(event.type, {...event}));
+	}
+
+	propChangedCallback ({name, prop, detail: change}) {
+		if (name === "spaces") {
+			this._el.picker.innerHTML = Object.entries(this.spaces)
+				.map(([id, space]) => `<option value="${id}">${space.name}</option>`)
+				.join("\n");
+		}
+
+		if (name === "value" || name === "spaces") {
+			if (this.value?.id !== this._el.picker.value) {
+				if (this.value.id in this.spaces) {
+					this._el.picker.value = this.value.id;
+				}
+				else {
+					let currentValue = this.spaces[this._el.picker.value];
+					console.warn(`No color space with id = “${ this.value.id }” was found among the specified ones. Using the current one (${ currentValue.id }) instead.`);
+					this.value = currentValue;
+				}
+			}
+		}
+	}
+
+	static props = {
+		value: {
+			default () {
+				return Self.Color.Space.get(this._el.picker.value);
+			},
+			parse (value) {
+				if (value instanceof Self.Color.Space || value === null || value === undefined) {
+					return value;
+				}
+
+				value += "";
+
+				return Self.Color.Space.get(value);
+			},
+			stringify (value) {
+				return value?.id;
+			},
+		},
+
+		spaces: {
+			type: {
+				is: Object,
+				get values () {
+					return Self.Color.Space;
+				},
+				defaultValue: (id, index) => {
+					try {
+						return Self.Color.Space.get(id);
+					}
+					catch (e) {
+						console.error(e);
+					}
+				},
+			},
+			default: () => Self.Color.spaces,
+			convert (value) {
+				// Drop non-existing spaces
+				let spaces = Object.fromEntries(Object.entries(value).filter(([id, space]) => space));
+				if (Object.keys(spaces).length === 0) {
+					console.warn("No color spaces with the specified ids were found. Using all the available ones instead.");
+					return Self.Color.spaces;
+				}
+
+				return spaces;
+			},
+			stringify (value) {
+				return Object.entries(value).map(([id, space]) => id).join(", ");
+			},
+		},
+	};
+
+	static events = {
+		change: {
+			from () {
+				return this._el.picker;
+			},
+		},
+		spacechange: {
+			propchange: "value",
+		},
+	};
+
+	static formAssociated = {
+		like: el => el._el.picker,
+		role: "combobox",
```

Comment:
Doesn't this default to the element we specified in `like`? If not, it probably should.

Reply:
[In our code, we try to default to `computedRole`](https://github.com/nudeui/element/blob/3c3297eed2506200998b43e29eaa0512c1f0bdcc/src/formAssociated.js/defineFormAssociated.js#L35). Unfortunately, it still isn't supported. It's probably under the flag in some browsers (not sure).

<img width="256" alt="image" src="https://github.com/user-attachments/assets/4ed1b55e-de8d-42e2-b124-6e10a328f6db">

Reply:
We should just use `getComputedAccessibleNode()` instead. It's not like we don't have access to this information at all.

Reply:
(in Nude Element, in case that wasn't clear)

Reply:
Got it. Thanks. I'll send a PR with the change.